File: docflow-frontend/types/api/AuditExportSnapshotCursorPageDTO.ts
import type { AuditExportSnapshotDTO } from "./AuditExportSnapshotDTO";

export type AuditExportSnapshotCursorPageDTO = {
  items: AuditExportSnapshotDTO[];
  hasMore: boolean;
  nextCursorCreatedAt: string | null; // ISO Instant
  nextCursorId: string | null; // UUID
};

File: docflow-backend/src/main/java/com/brutecx/docflow_backend/api/dto/audit/AuditExportSnapshotCursorPageDTO.java
package com.brutecx.docflow_backend.api.dto.audit;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

public record AuditExportSnapshotCursorPageDTO(
        List<AuditExportSnapshotDTO> items,
        boolean hasMore,
        Instant nextCursorTimestamp,
        UUID nextCursorId
) {
}
